src/main/scala/bot/domain/AdviseState.scala

```diff
@@ -0,0 +1,42 @@
+package bot.domain
+
+import bot.domain.buttons.CountryButtons._
+import bot.domain.buttons.WineTypeButtons._
+import vivino.domain.CountryCode._
+import vivino.domain.WineType._
+import vivino.domain.{CountryCode, WineType}
+
+case class AdviseState(
+  country: Option[String],
+  wineType: Option[String],
+  priceMin: Option[Int],
+  priceMax: Option[Int],
+) {
+  def toCountryCode: Option[CountryCode] = this.country.flatMap(AdviseState.CountryButtonToCountryCode.get)
+  def toWineType: Option[WineType] = this.wineType.flatMap(AdviseState.WineTypeButtonToWineType.get)
+}
+
+object AdviseState {
+  private val CountryButtonToCountryCode: Map[String, CountryCode] = Map(
+    (FranceCountryButton.name,       France),
```

Comment:
How to get a garantee on a type level that all CountryButtons are mentioned here?